These are still left:
```
../librz/arch/isa/arm/arm_il64.c:7:32: error: unknown option after ‘#pragma GCC diagnostic’ kind [-Werror=pragmas]
    7 | #pragma GCC diagnostic ignored "-Wenum-conversion"
      |                                ^~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
[1558/2286] Compiling C object librz/arch/librz_arch.so.0.8.0.p/isa_arm_armass64.c.o
[1559/2286] Compiling C object librz/arch/librz_arch.so.0.8.0.p/isa_arm_armass.c.o
[1560/2286] Compiling C object librz/arch/librz_arch.so.0.8.0.p/isa_arm_arm_il32.c.o
FAILED: librz/arch/librz_arch.so.0.8.0.p/isa_arm_arm_il32.c.o 
cc -Ilibrz/arch/librz_arch.so.0.8.0.p -I. -I.. -Ilibrz -I../librz -Ilibrz/include -I../librz/include -I../librz/arch/isa -I../librz/arch/isa_gnu -Ilibrz/arch -I../librz/arch -Ilibrz/util/sdb/src -I../librz/util/sdb/src -I../librz/bin/format -I../librz/type/parser -I../subprojects/capstone-next/include -I../subprojects/capstone-next/include/capstone -Isubprojects/rzspp -I../subprojects/rzspp -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Werror -O3 -Wimplicit-fallthrough=3 -DRZ_PLUGIN_INCORE=1 -DSUPPORTS_PCRE2_JIT -D_GNU_SOURCE -DCC_SUPPORTS_W_ENUM_COMPARE --std=gnu99 -Werror=sizeof-pointer-memaccess -fvisibility=hidden -Wno-cpp -fPIC -MD -MQ librz/arch/librz_arch.so.0.8.0.p/isa_arm_arm_il32.c.o -MF librz/arch/librz_arch.so.0.8.0.p/isa_arm_arm_il32.c.o.d -o librz/arch/librz_arch.so.0.8.0.p/isa_arm_arm_il32.c.o -c ../librz/arch/isa/arm/arm_il32.c
In file included from ../librz/arch/isa/arm/arm_il32.c:19:
../librz/arch/isa/arm/arm_accessors32.h:11:32: error: unknown option after ‘#pragma GCC diagnostic’ kind [-Werror=pragmas]
   11 | #pragma GCC diagnostic ignored "-Wenum-conversion"
      |                                ^~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
```

--- a/librz/arch/isa/arm/arm_accessors32.h
+++ b/librz/arch/isa/arm/arm_accessors32.h
@@ -7,8 +7,14 @@
  *     cs_insn *insn
  */
 
+#if CC_SUPPORTS_W_ENUM_COMPARE
 #pragma GCC diagnostic ignored "-Wenum-compare"
+#endif
+
+#ifdef CC_SUPPORTS_W_ENUM_CONVERION
 #pragma GCC diagnostic ignored "-Wenum-conversion"
+#endif
+
 #define CAPSTONE_AARCH64_COMPAT_HEADER
 #include <capstone/capstone.h>
 

--- a/librz/arch/isa/arm/arm_cs.h
+++ b/librz/arch/isa/arm/arm_cs.h
@@ -5,8 +5,15 @@
 #define RZ_ARM_CS_H
 
 #include <rz_analysis.h>
+
+#if CC_SUPPORTS_W_ENUM_COMPARE
 #pragma GCC diagnostic ignored "-Wenum-compare"
+#endif
+
+#ifdef CC_SUPPORTS_W_ENUM_CONVERION
 #pragma GCC diagnostic ignored "-Wenum-conversion"
+#endif
+
 #define CAPSTONE_AARCH64_COMPAT_HEADER
 #include <capstone/capstone.h>
 

--- a/librz/arch/isa/arm/arm_esil32.c
+++ b/librz/arch/isa/arm/arm_esil32.c
@@ -2,8 +2,15 @@
 // SPDX-License-Identifier: LGPL-3.0-only
 
 #include <rz_analysis.h>
+
+#if CC_SUPPORTS_W_ENUM_COMPARE
 #pragma GCC diagnostic ignored "-Wenum-compare"
+#endif
+
+#ifdef CC_SUPPORTS_W_ENUM_CONVERION
 #pragma GCC diagnostic ignored "-Wenum-conversion"
+#endif
+
 #define CAPSTONE_AARCH64_COMPAT_HEADER
 #include <capstone/capstone.h>
 

--- a/librz/arch/isa/arm/arm_esil64.c
+++ b/librz/arch/isa/arm/arm_esil64.c
@@ -3,8 +3,14 @@
 
 #include <rz_analysis.h>
 
+#if CC_SUPPORTS_W_ENUM_COMPARE
 #pragma GCC diagnostic ignored "-Wenum-compare"
+#endif
+
+#ifdef CC_SUPPORTS_W_ENUM_CONVERION
 #pragma GCC diagnostic ignored "-Wenum-conversion"
+#endif
+
 #define CAPSTONE_AARCH64_COMPAT_HEADER
 #include <capstone/capstone.h>
 

--- a/librz/arch/isa/arm/arm_il32.c
+++ b/librz/arch/isa/arm/arm_il32.c
@@ -4,8 +4,14 @@
 #include <rz_analysis.h>
 #include <rz_util/rz_assert.h>
 
+#if CC_SUPPORTS_W_ENUM_COMPARE
 #pragma GCC diagnostic ignored "-Wenum-compare"
+#endif
+
+#ifdef CC_SUPPORTS_W_ENUM_CONVERION
 #pragma GCC diagnostic ignored "-Wenum-conversion"
+#endif
+
 #define CAPSTONE_AARCH64_COMPAT_HEADER
 #include <capstone/capstone.h>
 

--- a/librz/arch/isa/arm/arm_il64.c
+++ b/librz/arch/isa/arm/arm_il64.c
@@ -3,8 +3,14 @@
 
 #include <rz_analysis.h>
 
+#if CC_SUPPORTS_W_ENUM_COMPARE
 #pragma GCC diagnostic ignored "-Wenum-compare"
+#endif
+
+#ifdef CC_SUPPORTS_W_ENUM_CONVERION
 #pragma GCC diagnostic ignored "-Wenum-conversion"
+#endif
+
 #define CAPSTONE_AARCH64_COMPAT_HEADER
 #include <capstone/capstone.h>
 

--- a/librz/arch/p/analysis/analysis_arm_cs.c
+++ b/librz/arch/p/analysis/analysis_arm_cs.c
@@ -5,8 +5,14 @@
 #include <rz_lib.h>
 #include <rz_util/ht_uu.h>
 
+#if CC_SUPPORTS_W_ENUM_COMPARE
 #pragma GCC diagnostic ignored "-Wenum-compare"
+#endif
+
+#ifdef CC_SUPPORTS_W_ENUM_CONVERION
 #pragma GCC diagnostic ignored "-Wenum-conversion"
+#endif
+
 #define CAPSTONE_AARCH64_COMPAT_HEADER
 #include <capstone/capstone.h>
 #include <capstone/arm.h>

--- a/librz/arch/p/analysis/analysis_sysz.c
+++ b/librz/arch/p/analysis/analysis_sysz.c
@@ -3,8 +3,15 @@
 
 #include <rz_analysis.h>
 #include <rz_lib.h>
+
+#if CC_SUPPORTS_W_ENUM_COMPARE
 #pragma GCC diagnostic ignored "-Wenum-compare"
+#endif
+
+#ifdef CC_SUPPORTS_W_ENUM_CONVERION
 #pragma GCC diagnostic ignored "-Wenum-conversion"
+#endif
+
 #define CAPSTONE_SYSTEMZ_COMPAT_HEADER
 #include <capstone/capstone.h>
 #include <capstone/systemz.h>

--- a/librz/arch/p/asm/asm_arm_cs.c
+++ b/librz/arch/p/asm/asm_arm_cs.c
@@ -4,8 +4,15 @@
 #include <rz_asm.h>
 #include <rz_lib.h>
 #include <rz_util/ht_uu.h>
+
+#if CC_SUPPORTS_W_ENUM_COMPARE
 #pragma GCC diagnostic ignored "-Wenum-compare"
+#endif
+
+#ifdef CC_SUPPORTS_W_ENUM_CONVERION
 #pragma GCC diagnostic ignored "-Wenum-conversion"
+#endif
+
 #define CAPSTONE_AARCH64_COMPAT_HEADER
 #include <capstone/capstone.h>
 #include "arm/asm-arm.h"

--- a/librz/arch/p/asm/cs_helper.h
+++ b/librz/arch/p/asm/cs_helper.h
@@ -3,8 +3,15 @@
 // SPDX-License-Identifier: LGPL-3.0-only
 
 #include <rz_util.h>
+
+#if CC_SUPPORTS_W_ENUM_COMPARE
 #pragma GCC diagnostic ignored "-Wenum-compare"
+#endif
+
+#ifdef CC_SUPPORTS_W_ENUM_CONVERION
 #pragma GCC diagnostic ignored "-Wenum-conversion"
+#endif
+
 #define CAPSTONE_SYSTEMZ_COMPAT_HEADER
 #include <capstone/capstone.h>
 

--- a/meson.build
+++ b/meson.build
@@ -79,6 +79,17 @@ if is_static_build
   endif
 endif
 
+# These two warnings enum compare and conversion are necessary for the Capstone compatibility header.
+# Newer compilers will warn about the comparison of different enums (arm64 and aarch64) in the header.
+# Hence, we have to disable them for the newer once.
+# Older compilers don't have these warnings and should not be included with #pragma
+if cc.has_argument('-Wenum-conversion')
+  add_global_arguments('-DCC_SUPPORTS_W_ENUM_CONVERION', language: ['c', 'cpp'])
+endif
+if cc.has_argument('-Wenum-compare')
+  add_global_arguments('-DCC_SUPPORTS_W_ENUM_COMPARE', language: ['c', 'cpp'])
+endif
+
 if cc.has_argument('--std=gnu99')
   add_global_arguments('--std=gnu99', language: ['c', 'cpp'])
 elif cc.has_argument('--std=c99')

--- a/subprojects/capstone-next.wrap
+++ b/subprojects/capstone-next.wrap
@@ -1,6 +1,6 @@
 [wrap-git]
 url = https://github.com/capstone-engine/capstone.git
-revision = 2cfca35e2fd2818d07b435e7ef551a723ffbf237
+revision = 306d5716d8faece2e6c74d5cf0a30be32e578aee
 directory = capstone-next
 patch_directory = capstone-next
 depth = 1

